src/js/controller/captions.js

```diff
@@ -42,36 +39,41 @@ const Captions = function(_model) {
         _tracks = [];
         _tracksById = {};
         _unknownCount = 0;
-    }
-
-    function _itemReadyHandler(item) {
-        // Clean up in case we're replaying
-        _itemHandler(_model, item);
 
+        const item = _model.get('playlistItem');
         const tracks = item.tracks;
         const len = tracks && tracks.length;
 
-        // Sideload tracks when not rendering natively
-        if (!_model.get('renderCaptionsNatively') && len) {
-            for (let i = 0; i < len; i++) {
-                /* eslint-disable no-loop-func */
-                const track = tracks[i];
-                if (_kindSupported(track.kind) && !_tracksById[track._id]) {
-                    _addTrack(track);
-                    loadFile(track,
-                        (vttCues) => {
-                            _addVTTCuesToTrack(track, vttCues);
-                        },
-                        (error) => {
-                            this.trigger(ERROR, {
-                                message: 'Captions failed to load',
-                                reason: error
-                            });
-                        });
+        // Update tracks once we know "renderCaptionsNatively" based on provider
+        _model.off('itemReady', null, this)
+            .once('itemReady', () => {
+                // Sideload tracks when not rendering natively
+                if (len && !_model.get('renderCaptionsNatively')) {
+                    for (let i = 0; i < len; i++) {
+                        /* eslint-disable no-loop-func */
+                        const track = tracks[i];
+                        if (_kindSupported(track.kind) && !_tracksById[track._id]) {
+                            _addTrack(track);
+                            loadFile(track,
+                                (vttCues) => {
+                                    _addVTTCuesToTrack(track, vttCues);
+                                },
+                                (error) => {
+                                    this.trigger(ERROR, {
+                                        message: 'Captions failed to load',
+                                        reason: error
+                                    });
+                                });
+                        }
+                        _updateMenu();
```

Comment:
Is there a reason behind updating the menu inside the for loop? Calling `_updateMenu()` after should set the menu up correctly, unless the desired behavior is an update for each track that gets added.

Reply:
This should be on the next like, outside of the loop for tracks added natively after the provider is inited (what other reason is there to wait for “itemReady”?)